hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -177,11 +179,14 @@ func (r *NodePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if controllerutil.ContainsFinalizer(nodePool, finalizer) {
 			controllerutil.RemoveFinalizer(nodePool, finalizer)
 			if err := r.Update(ctx, nodePool); err != nil {
+				metrics.StopRecording(nodePool)
```

Comment:
same questions than for HC.

Reply:
Same reply: because of a possible race condition.
Also note that the only reason we cannot remove a finalizer from a nodepool is when it has already been deleted... otherwise the `hypershift-operator` would have encountered other issues before.